### PR TITLE
Apply inclusion rules for political entities

### LINKED ISSUE #221: Missing British Overseas Territories
Came across [Akrotiri and Dhekelia](https://en.wikipedia.org/wiki/Akrotiri_and_Dhekelia) while looking something up. Any reason this territory of Britain is not included? It has a capital "Episkopi".

Then looking at the wiki page for [British Overseas Territories](https://en.wikipedia.org/wiki/British_Overseas_Territories) I saw that we also do not have:

- [Tristan da Cunha](https://en.wikipedia.org/wiki/Tristan_da_Cunha) (Capital: "Edinburgh of the Seven Seas" :sunglasses:)
- [South Georgia and the South Sandwich Islands](https://en.wikipedia.org/wiki/South_Georgia_and_the_South_Sandwich_Islands) (Capital: King Edward Point)

Every other British Overseas Territory seems to be accounted for :+1: 

###### Tbh I want Tristan da Cunha in the deck just because "Edinburgh of the Seven Seas" is fucking badass :laughing:

### LINKED ISSUE #306: Inclusion rules for political entities
As per https://github.com/axelboc/anki-ultimate-geography/issues/137#issuecomment-620764010, I'm opening this issue to try to define a set of strict inclusion rules for political entities.

### Sovereign states

This entity type has been well defined from the start thanks to Wikipedia's [List of sovereign states](https://en.wikipedia.org/wiki/List_of_sovereign_states). This list, sourced from the UN, includes 206 states. All 206 states are included in the deck, whether or not their sovereignty is disputed.

### Dependent territories

This entity type was first suggested in https://github.com/axelboc/anki-ultimate-geography/issues/221#issuecomment-606343365. [Quoting Wikipedia](https://en.wikipedia.org/wiki/Dependent_territory):

> A dependent territory, dependent area or dependency is a territory that does not possess full political independence or sovereignty as a sovereign state yet remains politically outside the controlling state's integral area.

Wikipedia's list of [depend territories](https://en.wikipedia.org/wiki/Dependent_territory#Lists_of_dependent_territories) is quite extensive as it includes uninhabited territories, only one of which is currently included in the deck (_British Indian Ocean Territory_). Adding all these territories is out of the question as per https://github.com/axelboc/anki-ultimate-geography/issues/137#issuecomment-620764010, so my recommendation is to refer to the list of [inhabited dependent territories](https://en.wikipedia.org/wiki/Dependent_territory#Overview_of_inhabited_dependent_territories) and remove _British Indian Ocean Territory_ from the deck.

I identified in [my audit](https://github.com/axelboc/anki-ultimate-geography/issues/221#issuecomment-607404670) that 36 of the 40 inhabited territories are already in the deck. The four that are not are: _Akrotiri and Dhekelia_, _Saint Pierre and Miquelon_, _Svalbard_, _Wallis and Futuna_, so these would need to be added leading to a net addition of 3 notes to the deck.

### Special overseas subdivisions

As per https://github.com/axelboc/anki-ultimate-geography/issues/137#issuecomment-620764010, countries' main subdivisions, like states and regions, are not to be included in the deck. Special subdivisions may however be included, particularly when they are located overseas. This should include, in my opinion:

- **French overseas regions**: _French Guiana_, _Guadeloupe_, _Martinique_, _Mayotte_, _Réunion_
- **Autonomous communities of Spain**: _Canary Islands_, _Ceuta_, _Melilla_, **[_Balearic Islands_](https://en.wikipedia.org/wiki/Balearic_Islands)** (not currently in the deck).
- **Special municipalities of the Netherlands**: _Bonaire_, _Saba_, _Sint Eustatius_
- **Autonomous regions of Portugal**: _Azores_, _Madeira_
- **Special territory of Chile**: _Easter Island_
- **Semi-autonomous region of Tanzania**: _Zanzibar_

The list may be extended in the future.

I recommend that the following subdivisions not be included in the deck as political entities (i.e. with flag and capital), as they do not have a special status: _Sardinia_ and _Sicily_ (regions of Italy), _Corsica_ (region of France), _Bali_ (province of Indonesia), _Galápagos Islands_ (province of Ecuador), _Kaliningrad Oblast_ (oblast of the Russian Federation). Some or all of them may remain in the deck as physical geography entities (i.e. with only a map).

### Special cases

In my opinion, the following entities are significant-enough politically to remain in the deck without the need for broader categorisation:

- **Constituent countries of the United Kingdom**: _England_, _Wales_, _Scotland_, _Northern Ireland_
- The **European Union**

The only remaining political entity is _Mount Athos_, which I recommend removing as it is really niche.

The deck also includes the following world regions: _Melanesia_, _Micronesia_, _Polynesia_, _Scandinavia_, _Balkan Peninsula_. I'm not quite sure whether these belong to the political or physical geography side of the deck... kind of neither? Perhaps they should be treated separately either way.

---

### Tentative actions summary

- Add _Akrotiri and Dhekelia_, _Saint Pierre and Miquelon_, _Svalbard_, _Wallis and Futuna_, and _Balearic Islands_.
- Remove _British Indian Ocean Territory_ and _Mount Athos_.
- Remove capitals and flags of _Sardinia_, _Sicily_, _Corsica_, _Bali_, _Galápagos Islands_, _Kaliningrad Oblast_ (but keep content of _Country info_ field).
- Document inclusion rules.

> Note that the changes above would grow the deck by three notes but shrink it by a few cards.

### PR DESCRIPTION
**Pull request by axelboc**
**UPDATE** - see https://github.com/axelboc/anki-ultimate-geography/pull/312#issuecomment-629833507 and following comments for final inclusion rules and removal stats.

---

Fixes #306, fixes #221.

Alright, so I've applied the latest inclusion rules discussed in #306. Here's what I've done exactly:

- Added the spreadsheet with the filtered lists at the root of the repository and renamed it to [political-entities.xlsx](https://github.com/axelboc/anki-ultimate-geography/files/4640132/political-entities.xlsx)
- Created a `temp` folder inside `src` with the following:
  - `data-temp.csv` - contains the data that moves out of the deck (I followed option D suggested by @aplaice in #307 without changing the GUIDs yet) -- note that I've added blank lines to group notes that are removed completely (first group) or only partially (second group).
  - `media` folder - contains the media files that move out of the deck.
  - `to-be-added.txt` - lists the notes that will need to be added later on to the deck.

The new inclusion rules would lead to the removal, in the standard deck, of:

- 21 location cards, or 6.5 % of the 321 cards currently in the deck. If we consider the 6 maps that would be added, this goes down to 15 cards or 4.7 %.
- 47 flag cards, or 18.1 % of the 260 cards currently in the deck.
- 104 capitals cards (= 2 templates * 52 capitals), or 19.9 % of the 524 cards currently in the deck.

**The net result would the removal of 165 cards, or 14.9 % of the 1105 cards currently in the deck.**

---

I have to admit that this is a **way more significant cull than I had envisioned**, sorry... TBH, I don't think removing more than 5~8 % of the deck is wise.

FWIW, I really like the result for autonomous islands, transcontinental areas and exclaves - i.e. _Ceuta_, _Saba_, etc. being removed completely, and _Corsica_, _Sicily_, _Galápagos Islands_, etc. having their capitals and flags removed.

However, I think it would be wiser to **rethink the rules for dependent territories**. Here is one suggestion:

- include all of them with maps;
- apply the `OR` rule for inclusion with capitals/flags (potentially tweaking the area/population thresholds);
- remove the now-obsolete `AND` rule.

I think the above brings some nice benefits (i.e. exhaustiveness, while still removing some of the most obscure capitals and flags), and would not have such a strong impact on the deck.

What are your thoughts?

### CODE REVIEW

**Comment by aplaice** [comment]
Wow!

I understand that we'll wait with the (probable) full removal of entities like Mount Athos or the (probable) removal of flag/capital from Bali, for a future analysis of non-island autonomous countries and of non-autonomous islands (as physical entities)?

<hr/>

> I have to admit that this is a way more significant cull than I had envisioned, sorry... TBH, I don't think removing more than 5~8 % of the deck is wise.

Do you mean removing more than 5-8 % of the current cards, or on net? If the latter, then we can easily (and sensibly) fill up the gap with islands (as non-political entities) and potentially regions, deserts, lakes, mountain ranges, non-island autonomous regions or even some more seas. :D

> apply the OR rule for inclusion with capitals/flags (potentially tweaking the area/population thresholds);

On a gut level, I care far more about and are far more likely to ever encounter, in real life, the capitals (and even flags) of Sicily, Corsica, Bali (non-autonomous that it is) or Zanzibar than of, say Jersey (which is among the "marginal" dependent territories likely to get the capital/flag), irrespective of their political status. I would guess that that might hold for many people. That isn't to say that I want the capitals of the former kept, but I'm also not enthusiastic about re-adding (keeping) them for the latter group.

> include all of them with maps;

I feel slightly less opposed to including all the dependent territories with maps, for exhaustiveness, though now that I look at them, some are absurdly small (e.g. Pitcairn Islands, pop. < 60)...

Hence, my alternative suggestion:

1. Include all the dependent territories with maps.

2. Get rid of the now obsolete OR criterion.

However, I'd also be happy with the full cull.

<hr/>

FWIW I haven't yet thoroughly looked at the commit, but on a cursory glance, it looks great implementation-wise!

**Comment by axelboc** [comment]
> I understand that we'll wait with the (probable) full removal of entities like Mount Athos or the (probable) removal of flag/capital from Bali, for a future analysis of non-island autonomous countries and of non-autonomous islands (as physical entities)?

Ha ! No, I just forgot about them. 😄 I've removed _Mount Athos_ completely and removed Bali's capital and flag for now. I've also changed Bali's capital info to "Island of Indonesia" and updated the numbers in the PR description.

**Comment by axelboc** [comment]
> Do you mean removing more than 5-8 % of the current cards, or on net? If the latter, then we can easily (and sensibly) fill up the gap with islands (as non-political entities) and potentially regions, deserts, lakes, mountain ranges, non-island autonomous regions or even some more seas. :D

I did mean net, but only within the political geography side of the deck. Adding more physical geography notes is not going to happen in the short term, so I prefer not to assume too much for now. 😄 Basically, I'm all for removing a bunch of cards, but I don't want it to be a frustrating experience for our users, you know? 

You make good points about some dependent territories being very small in terms of population. I went back to the spreadsheet to play with the criteria a bit, and try to get more sensible results... I think I managed to keep a bit more notes and a bit more capitals and flags, while not going so far as to include maps for every dependent territory:

[political-entities.xlsx](https://github.com/axelboc/anki-ultimate-geography/files/4640656/political-entities.xlsx)

Here's an overview:

- Include dependent territories with map if `population >= 15,000 OR area >= 1,000` -- this excludes 10 minor territories but keeps _Svalbard_ and _Falkland Islands_.
- Include dependent territories with capital and flag if `population >= 100,000 OR (population >= 15,000 AND area >= 1,000)` -- this includes 9 highly populated territories, as well as _Greenland_, _Faroe Islands_ and _Åland Islands_, but excludes _Jersey_ (118 km2) and _Svalbard_ (2,667 inhabitants) notably.
- Include autonomous island with map if `population >= 100,000` -- I realised that _Madeira_ and _Mayotte_, which both have around 280,000 inhabitants were being excluded because of their area; I think applying a high population threshold on its own makes more sense.
- Include transcontinental territories and exclaves with map if `population >= 100,000 AND distance >= CLOSE` -- similarly, I think removing the area threshold makes sense; note that this removes _Socotra Governorate_ and _Galápagos Islands_.

What do you think?

**Comment by aplaice** [comment]
This is great!

Its one disadvantage is obviously that it's more complicated, but I think the thresholds make general sense, and I really like the result! Since the various thresholds will be in the spreadsheet, in the repo, the additional complexity isn't likely to be an issue.

**Comment by aplaice** [comment]
Unfortunately, I won't have time to properly look through this until next week, but on a cursory glance I haven't seen any issues.

**Comment by aplaice** [comment]
This looks great!

I haven't found issues in any of the plausible categories (overall implementation of the spreadsheet, fields changed where they shouldn't be or not changed where they should be, moving of the images).


## Capital hints in removed cards

One quibble I have is the removal of the capital hints from `src/temp/data-temp.csv`.  For the countries which are "fully" staying in `src/data.csv`, but whose capital "pair" has been removed (fully or partially), and hence whose capital hints are not needed, the removal of the hints makes sense.  (Temporarily keeping the hints for the benefit of upgrading users who aren't necessarily aware which notes/cards were removed (and hence don't initally realise that, say, the capital *Kingston* only corresponds to a single country), might have been useful, but it'd have come at the unacceptable cost of confusing new users.)  However, keeping the capital hints in the removed cards would have some advantages for the users who want to keep using them:

1. (Assuming that the user keeps the "removed" cards separate.) In principle, the fact that the cards are in the separate deck is a hint on its own (i.e. if you see *Kingston* on a Capital->Country card, in the "removed" deck, you know that the corresponding "country" has to be a dependent territory, island or autonomous region, so you don't need the previous hint for *Norfolk Island* saying "Not a sovereign country"). However, users who haven't been keeping track of this issue won't necessarily realise this. This is particularly salient for the *Sint Eustatius*/*Aruba* pair, since neither is a sovereign country and hence without having closely followed the details of the issue, the user won't know which one was removed and which one wasn't, but I think it also holds to some extent for the other cases. Obviously (as noted), keeping the hint for both elements of the pairs is not an option, but IMO keeping it for only one of them is still better than not having it for either.

2. (Assuming that the user keeps the "removed" cards together with the main deck, for instance because they don't have a dedicated AUG deck and have all the cards in a more-general "Misc" or "Geography" deck, because they cheekily move the "removed" cards back, or because they use custom studies with mixed decks.) In this case, the Capital hint would be essential for disambiguation. As in 1. having the Capital hint for only one element of each pair would still be valuable, as it'd be technically sufficient for disambiguation purposes.

The disadvantage would be that the users who keep the removed cards, separately, would arguably have an excessive hint. However, as argued above I don't think it'd be excessive and besides it's a hint that had already been present.

Obviously, new/fresh users would not be adversely affected.

Hence, overall, I'd suggest keeping the capital hints in the removed cards, if possible.

**Comment by axelboc** [comment]
Makes total sense, and it's consistent with keeping the flag similarities in the removed cards. I'll put the capital hints back in then asap, then 👍

File: src/data.csv [hunk 1]
```diff
@@ -159,13 +159,13 @@ q1+V%/.{Ud,Brazil,Brasilien,Brasil,Brésil,Brasil,,,,,,Brasilia,Brasília,Brasil
...
 jY<1R8gh$5,Ecuador,Ecuador,Ecuador,Équateur,Ecuador,,,,,,Quito,Quito,Quito,Quito,Quito,,,,,,,,,,,"<img src=""ug-flag-ecuador.png"" />","Colombia (no coat of arms)","Kolumbien (kein Wappen)","Colombia (sin escudo)","Colombie (pas de blason)","Colombia (uten våpenskjold)","<img src=""ug-map-ecuador.png"" />","UG::South_America UG::Sovereign_State"
-ow]5&Kht`e,Guyana,Guyana,Guyana,Guyana,Guyana,,,,,,Georgetown,Georgetown,Georgetown,Georgetown,Georgetown,,,,,,"Sovereign country","Souveräner Staat","Es un país soberano","Nation souveraine","Selvstendig land","<img src=""ug-flag-guyana.svg"" />",,,,,,"<img src=""ug-map-guyana.png"" />","UG::South_America UG::Sovereign_State"
+ow]5&Kht`e,Guyana,Guyana,Guyana,Guyana,Guyana,,,,,,Georgetown,Georgetown,Georgetown,Georgetown,Georgetown,,,,,,,,,,,"<img src=""ug-flag-guyana.svg"" />",,,,,,"<img src=""ug-map-guyana.png"" />","UG::South_America UG::Sovereign_State"
```
[the remark this answers is not in the document]
Comment (axelboc) [reply src/data.csv:162]: Capital hint re-added to `data-temp.csv`.

File: src/data.csv [hunk 2]
```diff
@@ -176,10 +176,10 @@ rcI^]YY+!g,"Puerto Rico","Puerto Rico","Puerto Rico","Puerto Rico","Puerto Rico"
...
 f}r?r8kW=^,Haiti,Haiti,Haití,Haïti,Haiti,,,,,,Port-au-Prince,Port-au-Prince,"Puerto Príncipe",Port-au-Prince,Port-au-Prince,,,,,,,,,,,"<img src=""ug-flag-haiti.svg"" />",,,,,,"<img src=""ug-map-haiti.png"" />","UG::Caribbean UG::North_America UG::Sovereign_State"
-fnDvVbeE_Z,Jamaica,Jamaika,Jamaica,Jamaïque,Jamaica,,,,,,Kingston,Kingston,Kingston,Kingston,Kingston,,,,,,"Sovereign country","Souveräner Staat","Es un país soberano","Nation souveraine","Selvstendig land","<img src=""ug-flag-jamaica.svg"" />",,,,,,"<img src=""ug-map-jamaica.png"" />","UG::Caribbean UG::North_America UG::Sovereign_State"
+fnDvVbeE_Z,Jamaica,Jamaika,Jamaica,Jamaïque,Jamaica,,,,,,Kingston,Kingston,Kingston,Kingston,Kingston,,,,,,,,,,,"<img src=""ug-flag-jamaica.svg"" />",,,,,,"<img src=""ug-map-jamaica.png"" />","UG::Caribbean UG::North_America UG::Sovereign_State"
```
[the remark this answers is not in the document]
Comment (axelboc) [reply src/data.csv:179]: Capital hint re-added to `data-temp.csv`.

File: src/data.csv [hunk 3]
```diff
@@ -193,72 +193,57 @@ kb%EK<T(Tt,"Antigua and Barbuda","Antigua und Barbuda","Antigua y Barbuda",Antig
...
-~Dk{D{2gBH,"Saint Kitts and Nevis","St. Kitts und Nevis","San Cristóbal y Nieves",Saint-Christophe-et-Niévès,"Saint Kitts og Nevis",,,,,,Basseterre,Basseterre,Basseterre,Basseterre,Basseterre,,,,,,"Sovereign country","Souveräner Staat","Es un país soberano","Nation souveraine","Selvstendig land","<img src=""ug-flag-saint_kitts_and_nevis.svg"" />",,,,,,"<img src=""ug-map-saint_kitts_and_nevis.png"" />","UG::Caribbean UG::North_America UG::Sovereign_State"
+~Dk{D{2gBH,"Saint Kitts and Nevis","St. Kitts und Nevis","San Cristóbal y Nieves",Saint-Christophe-et-Niévès,"Saint Kitts og Nevis",,,,,,Basseterre,Basseterre,Basseterre,Basseterre,Basseterre,,,,,,,,,,,"<img src=""ug-flag-saint_kitts_and_nevis.svg"" />",,,,,,"<img src=""ug-map-saint_kitts_and_nevis.png"" />","UG::Caribbean UG::North_America UG::Sovereign_State"
```
[the remark this answers is not in the document]
Comment (axelboc) [reply src/data.csv:226]: Capital hint re-added to `data-temp.csv`.

File: src/data.csv [hunk 4]
```diff
@@ -193,72 +193,57 @@ kb%EK<T(Tt,"Antigua and Barbuda","Antigua und Barbuda","Antigua y Barbuda",Antig
...
+nK;HFcd-1P,"Northern Mariana Islands","Nördliche Marianen","Islas Marianas del Norte","Îles Mariannes du Nord",Nord-Marianene,"Unincorporated territory of the United States.","Nichtinkorporiertes US-amerikanisches Außengebiet.","Territorio no incorporado de los Estados Unidos.","Territoire non incorporé des États-Unis.","Uinnlemmet territorium i USA.",,,,,,,,,,,,,,,,,,,,,,"<img src=""ug-map-northern_mariana_islands.png"" />",UG::Oceania
+s7{M&?)+C&,Aruba,Aruba,Aruba,Aruba,Aruba,"Constituent country of the Kingdom of the Netherlands.","Autonomes Land innerhalb des Königreichs der Niederlande.","Nación constitutiva del Reino de los Países Bajos.","État autonome du royaume des Pays-Bas.","«Land» som sammen med Nederland, Curaçao og Sint Maarten utgjør føderasjonen Kongeriket Nederlandene.",Oranjestad,Oranjestad,Oranjestad,Oranjestad,Oranjestad,,,,,,,,,,,"<img src=""ug-flag-aruba.svg"" />",,,,,,"<img src=""ug-map-aruba.png"" />","UG::Caribbean UG::North_America"
```
[the remark this answers is not in the document]
Comment (axelboc) [reply src/data.csv:236]: Capital hint re-added to `data-temp.csv`.

**Comment by axelboc** [comment]
I've updated `sources.csv`, double checked the removed media, moved the `temp` folder to the root of the repo and renamed it `removed` .. but most importantly, I've documented the latest inclusion rules in `CONTRIBUTING.md`. Please do proof-read if you have time! 🙏

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -54,6 +57,70 @@ Anki requires each note to have a unique identifier. When you add a note to the
...
+
+- Source: https://en.wikipedia.org/wiki/List_of_autonomous_areas_by_country
+- Specifically insular autonomous territories that are not also sovereign states or dependent territories.
+- Criteria for inclusion with map: `population >= 100,000`
+- Capitals and flags never included.
+
+#### Transcontinental overseas territories
+
+- Source: https://en.wikipedia.org/wiki/List_of_transcontinental_countries
+- Specifically inhabited territories located on a different content than their mainland, and that are not also dependent territories.
```
Comment (aplaice) [inline CONTRIBUTING.md:95]: If we don't go with a single summary at the start, we should probably also mention "not an autonomous island". (Mayotte is both an autonomous island and transcontinental.)

**Comment by aplaice** [comment]
I've just realised that the Cook Islands and Niue are both in the ["sovereign countries" list](https://en.wikipedia.org/wiki/List_of_sovereign_states), so they should both be fully included...

**Comment by axelboc** [comment]
Well spotted for Niue and Cook Islands!! 👓 

You're totally right about prioritising entity types. I've done as you suggested:

- clarified `CONTRIBUTING.md`,
- removed Niue and Cook Islands from the list of dependent territories,
- moved French Guiana to the list of transcontinental territories,
- removed Ceuta and Melilla from the list of enclaves/exclaves.

**Comment by aplaice** [comment]
I don't see any remaining issues and I don't think we'll catch anything more, (and even if we do, we can just fix it on master), so I'd vote for merging!

<hr/>

Apparently even people who are dedicated enough to file issues about the flag of the Northern Mariana Islands don't mind its removal, so I think we're definitely doing "the right thing", even if a 12 % removal is more than what I'd have expected at the very start.

**Comment by axelboc** [comment]
Yeah, it's reassuring 😄 

Alright, let's do this!! 🤘

**Pull request merged**